### ISSUE TITLE
amplify init: -unauthRole doesn't exist

### ISSUE THREAD

**Issue by mrdavidhanson** [aws-amplify/amplify-cli#353]
**Describe the bug**
Running "amplify init" results in this error:

Template error: IAM role ampstarter-20181025120808-unauthRole doesn't exist.

**amplify --version**
0.1.29

**region**
us-east-1

**To Reproduce**
Steps to reproduce the behaviour:
1. npx create-react-app ampstarter

2. cd ampstarter 

3. amplify init
settings: 
- Choose your default editor: Visual Studio Code
- Choose the type of app that you're building javascript
Please tell us about your project
- What javascript framework are you using react
- Source Directory Path:  src
- Distribution Directory Path: build
- Build Command:  npm run-script build
- Start Command: npm run-script start
- Do you want to use an AWS profile? Yes
- Please choose the profile you want to use default
Using default provider awscloudformation

4. see output log below

**Desktop:**
 - OS: macOS High Sierra 10.13.6

**Log**
⠏ Initializing project in the cloud...

CREATE_IN_PROGRESS ampstarter-20181025120808 AWS::CloudFormation::Stack Thu Oct 25 2018 12:08:08 GMT-0400 (Eastern Daylight Time) User Initiated
CREATE_IN_PROGRESS UnauthRole                AWS::IAM::Role             Thu Oct 25 2018 12:08:12 GMT-0400 (Eastern Daylight Time)
CREATE_IN_PROGRESS AuthRole                  AWS::IAM::Role             Thu Oct 25 2018 12:08:12 GMT-0400 (Eastern Daylight Time)
CREATE_IN_PROGRESS UnauthRole                AWS::IAM::Role             Thu Oct 25 2018 12:08:12 GMT-0400 (Eastern Daylight Time) Resource creation Initiated
CREATE_IN_PROGRESS AuthRole                  AWS::IAM::Role             Thu Oct 25 2018 12:08:13 GMT-0400 (Eastern Daylight Time) Resource creation Initiated
CREATE_IN_PROGRESS DeploymentBucket          AWS::S3::Bucket            Thu Oct 25 2018 12:08:13 GMT-0400 (Eastern Daylight Time)
CREATE_IN_PROGRESS DeploymentBucket          AWS::S3::Bucket            Thu Oct 25 2018 12:08:14 GMT-0400 (Eastern Daylight Time) Resource creation Initiated
⠼ Initializing project in the cloud...

CREATE_COMPLETE AuthRole   AWS::IAM::Role Thu Oct 25 2018 12:08:25 GMT-0400 (Eastern Daylight Time)
CREATE_COMPLETE UnauthRole AWS::IAM::Role Thu Oct 25 2018 12:08:26 GMT-0400 (Eastern Daylight Time)
⠦ Initializing project in the cloud...

CREATE_COMPLETE      DeploymentBucket          AWS::S3::Bucket            Thu Oct 25 2018 12:08:34 GMT-0400 (Eastern Daylight Time)
ROLLBACK_IN_PROGRESS ampstarter-20181025120808 AWS::CloudFormation::Stack Thu Oct 25 2018 12:08:36 GMT-0400 (Eastern Daylight Time) Template error: IAM role ampstarter-20181025120808-unauthRole doesn't exist. Rollback requested by user.
⠧ Initializing project in the cloud...

DELETE_IN_PROGRESS UnauthRole       AWS::IAM::Role  Thu Oct 25 2018 12:08:46 GMT-0400 (Eastern Daylight Time)
DELETE_IN_PROGRESS AuthRole         AWS::IAM::Role  Thu Oct 25 2018 12:08:46 GMT-0400 (Eastern Daylight Time)
DELETE_SKIPPED     DeploymentBucket AWS::S3::Bucket Thu Oct 25 2018 12:08:46 GMT-0400 (Eastern Daylight Time)
DELETE_COMPLETE    UnauthRole       AWS::IAM::Role  Thu Oct 25 2018 12:08:47 GMT-0400 (Eastern Daylight Time)
DELETE_COMPLETE    AuthRole         AWS::IAM::Role  Thu Oct 25 2018 12:08:47 GMT-0400 (Eastern Daylight Time)
⠦ Initializing project in the cloud...

ROLLBACK_COMPLETE ampstarter-20181025120808 AWS::CloudFormation::Stack Thu Oct 25 2018 12:08:49 GMT-0400 (Eastern Daylight Time)
⠴ Initializing project in the cloud...An error occurred when  creating the CloudFormation stack
✖ Root stack creation failed
init failed
{ ResourceNotReady: Resource is not in the state stackCreateComplete
    at constructor.setError (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/node_modules/aws-sdk/lib/resource_waiter.js:182:47)
    at Request.CHECK_ACCEPTORS (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/node_modules/aws-sdk/lib/resource_waiter.js:44:12)
    at Request.callListeners (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/node_modules/aws-sdk/lib/sequential_executor.js:106:20)
    at Request.emit (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/node_modules/aws-sdk/lib/request.js:683:14)
    at Request.transition (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /usr/local/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/node_modules/aws-sdk/lib/request.js:685:12)
  message: 'Resource is not in the state stackCreateComplete',
  code: 'ResourceNotReady',
  retryable: false,
  time: 2018-10-25T16:09:11.803Z,
  statusCode: 200,
  retryDelay: 30000 }

**Comment by parachut-daniel** [comment]
I don't know if this helps, however this issue arose after I "amplify delete"'`d on a react-native project. Although it seems to not be a project specific issue, fresh installs don't seem to work either.

**Comment by UnleashedMind** [comment]
Could you try the the following steps, see if it resolves the issue: 
1. check that no aws services has reached its limit
2. remove all amplify-cli artifacts on the local project.
3. execute `amplify init` again, and when prompted, enter a different project name that's never been used before, the "projectName" is used to prefix resource names, use a different name is to avoid name clash in aws.

**Comment by adamup928** [comment]
I can confirm this same issue. After following those three steps, I'm not prompted to enter a project name.  I don't see any remaining (visible or hidden files) artifacts in the directory.  `amplify init` still fails

**Comment by d3v0purr** [comment]
Thanks for all of your replies!

Here is a screenshot from the AWS Console of the Events for the corresponding failed Cloud Formation Stack that amplify init generated:

<img width="1232" alt="stack_detail" src="https://user-images.githubusercontent.com/447305/47521681-a0713080-d861-11e8-91d4-b8dceb82fe03.png">

@UnleashedMind - 

My AWS account has not hit any limits. I encountered this issue at least three times this morning while attempting to troubleshoot the issue - each time was with a fresh project with a different name. 

**Success**
Oddly enough, I just tried to reproduce the problem again with another fresh project and this time was successful! I notice the timing of the events in the Cloud Formation Stack are slightly different. Here are the logs for the successful Cloud Formation Stack:

<img width="1220" alt="stack_detail-success" src="https://user-images.githubusercontent.com/447305/47522416-87697f00-d863-11e8-8345-0484604889d6.png">

**Another Failure**
After obtaining success, I tried again with another fresh project -- this time, another failure. Same failure reason. Here is another screenshot:

<img width="1196" alt="stack_detail-ampstarter4-fail" src="https://user-images.githubusercontent.com/447305/47523066-280c6e80-d865-11e8-9f2c-8b23b822ef41.png">
